src/k2/transport/Payload.cpp

```diff
@@ -194,19 +194,21 @@ bool Payload::read(String& value) {
     return read((void*)value.data(), size);
 }
 
-bool Payload::read(std::decimal::decimal64& value) {
-    std::decimal::decimal64::__decfloat64 data;
-    bool success = read((void*)&data, sizeof(data));
+bool Payload::read(boost::multiprecision::cpp_dec_float_50& value) {
+    Binary binData;
+    bool success = read(binData);
     if (!success) return false;
-    value.__setval(data);
+    boost::multiprecision::cpp_dec_float_50 val(binData.get());
+    value = val;
```

Comment:
combine these in one line to make more efficient: value = boost::multiprecision::cpp_dec_float_50 val(binData.get());

Also, according to the boost API, this can throw. You need to catch std::exception and convert to false return value

Reply:
Changes have been pushed here:
https://github.com/futurewei-cloud/chogori-platform/compare/master...change-decimal-library